src/vs/workbench/browser/parts/editor/editorActionBarControl.tsx

```diff
@@ -262,23 +243,120 @@ export class EditorActionBarControlFactory {
 	dispose(): void {
 		this._disposables.dispose();
 		this._controlDisposables.dispose();
+		this._editorDisposables.dispose();
 	}
 
 	//#endregion Constructor & Dispose
 
 	//#region Private Methods
 
 	/**
-	 * Creates the control.
-	 * @returns The control.
+	 * Sets up the editor.
+	 * @param editorInput The editor input.
 	 */
-	private createControl() {
-		// Create the control.
-		this._control = this._controlDisposables.add(this._instantiationService.createInstance(
-			EditorActionBarControl,
-			this._container,
-			this._editorGroup
-		));
+	private setupEditor(editorInput: EditorInput | undefined) {
+		// Dispose of the previous editor disposables.
+		this._editorDisposables.clear();
+
+		// If there isn't an active editor, disable the editor action bar and return.
+		if (!editorInput) {
+			this.updateEnablement(false);
+			return;
+		}
+
+		// Data Explorer always enables editor action bar.
+		if (editorInput.typeId === PositronDataExplorerEditorInput.TypeID) {
+			this.updateEnablement(true);
+			return;
+		}
+
+		// Settings always enables editor action bar.
+		if (editorInput.typeId === SettingsEditor2Input.ID) {
+			this.updateEnablement(true);
+			return;
+		}
+
+		// Get the active editor pane. If there isn't one, disable the editor action bar and return.
+		const activeEditorPane = this._editorGroup.activeEditorPane;
+		if (!activeEditorPane) {
+			this.updateEnablement(false);
+			return;
+		}
+
+		// Get editor control. If it's not code editor, disable the editor action bar and return.
+		const editorControl = activeEditorPane.getControl();
+		if (!isCodeEditor(editorControl)) {
+			this.updateEnablement(false);
+			return;
+		}
+
+		// Get the text model. If there isn't one, disable the editor action bar and return.
+		const textModel = editorControl.getModel();
+		if (!textModel) {
+			this.updateEnablement(false);
+			return;
+		}
+
+		// Update the enablement based on the text model's language.
+		this.updateEnablementForLanguageAndOptions(textModel.getLanguageId(), editorControl.getOption(EditorOption.actionBar));
+
+		// Add a listener for language changes on the text model. This ensures enablement is updated
+		// when the language changes, which is especially important when Positron restores editors
+		// before language extensions have fully loaded.
+		this._editorDisposables.add(textModel.onDidChangeLanguage(e => {
+			this.updateEnablementForLanguageAndOptions(textModel.getLanguageId(), editorControl.getOption(EditorOption.actionBar));
+		}));
+
+		// Add a listener for configuration changes on the editor control. This ensures enablement is
+		// updated when the configuration changes.
+		this._editorDisposables.add(editorControl.onDidChangeConfiguration(e => {
+			this.updateEnablementForLanguageAndOptions(textModel.getLanguageId(), editorControl.getOption(EditorOption.actionBar));
+		}));
+	}
+
+	/**
+	 * Udpates the enablement for the specified language.
+	 * @param language
+	 */
```

Comment:
It looks like this comment has a type and is incomplete. We could update it to something like this:

```suggestion
	/**
	 * Creates or destroys the editor action bar for the provided language based off user preferences
	 * @param language The language of the file in the editor pane
	 * @ editorActionBarOptions The user preferences for the editor action bar
	 * @ editorActionBarOptions. hiddenForLanguages The languages the editor action bar should be hidden for
	 */
```